### DISCUSSION TITLE
line height classes do not work

### DISCUSSION THREAD

**Discussion by uurcank** [tailwindlabs/tailwindcss#5977] [Help]
**What version of Tailwind CSS are you using?**

For example: v2.2.0 (latest)

**What build tool (or framework if it abstracts the build tool) are you using?**

Rails 6.1 with Tailwind JIT

**What browser are you using?**

Chrome latest

**What operating system are you using?**

Windows
 
**Describe your issue**

leading-loose and all other line-height classes do not work.  text-xl class contains "line-height" property and it overrides line-height classes. It is expected line-height classes to override the property in font-size classes

**Comment by adamwathan** [comment]
They definitely work:

https://play.tailwindcss.com/gym6S1c3y4

...so it must be something specific to your project. Please provide a reproduction to get the most useful help.

**Reply by hyvyys** [reply]
They don't work if you use a breakpoint class like `md:text-lg leading-loose`, you then also have to use `md:leading-loose`. A design flaw IMHO.

**Reply by fturcheti** [reply]
> They don't work if you use a breakpoint class like `md:text-lg leading-loose`, you then also have to use `md:leading-loose`. A design flaw IMHO.

That's exactly the problem. If we have something like `text-lg md:text-xl lg:text-2xl` and would like to set it to `leading-tight`, we'd have to set `leading-tight` to each breakpoint, `leading-tight md:leading-tight lg:leading-tight`, or use a "important modifier", `!leading-tight`, because the `line-height` in each `font-size` breakpoint overwrites the `line-height` set without a breakpoint variant.

I find this to be counter intuitive and something that goes against the principle of having single-purposes classes. Maybe this should be an opt-in feature: if you want, you can define line-heights with your font-sizes; but I think this should not be the default. By default we should have font-sizes and line-heights working separately.

**Reply by uurcank** [reply]
@adamwathan  this should be converted to an issue

**Reply by adamwathan** [reply]
This works the way we want it to, I would never want the line-height to be the same when the font size changes, even if it's defined using a relative value. Larger fonts need tighter line-heights than smaller fonts to look good. 16px with 1.5 line-height is 24px line-height, but 32px font with 1.5 line-height is 48px which is too much. So even if Tailwind didn't reset the line-height to a new value when you change the font size at a different breakpoint, you should be setting new line-height value yourself anyways because the line-height you were using for the previous font-size is almost guaranteed to be inappropriate for the new font-size.

So the current behavior is how I personally want it to work and no plans to change it, sorry if we disagree on this one.

If you want this to behave differently in your project, just update your config so your font size scale doesn't include any default line-height values:

https://play.tailwindcss.com/pPAjnYuede?file=config

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    fontSize: {
      xs: '0.75rem',
      sm: '0.875rem',
      base: '1rem',
      lg: '1.125rem',
      xl: '1.25rem',
      '2xl': '1.5rem',
      '3xl': '1.875rem',
      '4xl': '2.25rem',
      '5xl': '3rem',
      '6xl': '3.75rem',
      '7xl': '4.5rem',
      '8xl': '6rem',
      '9xl': '8rem',
    },
  },
  plugins: [],
}
```

**Comment by yogeshnogia** [comment]
Try using safelist in config file, you can paste any class here, for eg 'leading-7'
`safelist: [
    'bg-red-500',
    'text-3xl',
    'lg:text-4xl',
    'leading-7',
  ]`

**Comment by NickWoodward** [comment]
Just on the off chance someone comes here and has an issue with leading-* classes not being applied to the dom, for me it was that I was using arbitrary sizes, which meant that the order of the text-size and leading values mattered.

doesn't work:
`leading-8 text-[26px]`

does:
`text-[26px] leading-8`

this may look obvious above, but confused me temporarily when using the `cn()` function to pass only the arbitrary size, ie:

Doesn't work:
```
<Test className="text-[26]">

export const Test = (className) => {
  return <div className={cn("leading-8", className)}></div>
}
```

**Comment by vitpankin** [comment]
Just caught the same issue, leading does not work if it's first in the classname string

**Comment by gpm-developer** [comment]
the only solution I found is using style={{ lineHeight: <"value">}}, works perfect tho, why tailwind team hasn't fixed this already?
